test/e2e/util/util.go

```diff
@@ -64,16 +65,16 @@ func CreateCluster(t *testing.T, ctx context.Context, client crclient.Client, op
 
 	// Define a standard teardown function
 	teardown := func(ctx context.Context) {
-		SaveMachineConsoleLogs(t, ctx, hc, opts.AWSCredentialsFile, artifactDir)
+		SaveMachineConsoleLogs(t, ctx, hc, opts.AWSPlatform.AWSCredentialsFile, artifactDir)
 		// TODO: Figure out why this is slow
 		//e2eutil.DumpGuestCluster(context.Background(), client, hostedCluster, globalOpts.ArtifactDir)
-		DumpAndDestroyHostedCluster(t, ctx, hc, opts.AWSCredentialsFile, opts.Region, opts.BaseDomain, artifactDir)
+		DumpAndDestroyHostedCluster(t, ctx, hc, opts.AWSPlatform.AWSCredentialsFile, opts.AWSPlatform.Region, opts.AWSPlatform.BaseDomain, artifactDir)
 		DeleteNamespace(t, ctx, client, namespace.Name)
 	}
 
 	// Try and create the cluster
 	t.Logf("Creating a new cluster. Options: %v", opts)
-	if err := cmdcluster.CreateCluster(ctx, opts); err != nil {
+	if err := aws.CreateCluster(ctx, opts); err != nil {
```

Comment:
note: At some point we'll need to make this dynamic, possibly having an interface and an passing through the platform specific implementation.
We'll get there when enabling e2e for other platforms

Reply:
Definitely. I have a wip for e2e testing. Will post it once this one is merged.